src/vzlogger.cpp

```diff
@@ -135,26 +137,32 @@ void print(log_level_t level, const char *format, const char *id, ...) {
 	if (getppid() != 1) { /* running as fork in background? */
 		FILE *stream = (level > 0) ? stdout : stderr;
 
+		m_log.lock(); // safe write access for competed access from other thread
 		fprintf(stream, "%-24s", prefix);
 		vfprintf(stream, format, args);
 		fprintf(stream, "\n");
+		m_log.unlock(); // release mutex
 	}
 	va_end(args);
 
 	va_start(args, id);
 	/* append to logfile */
 	if (options.logfd()) {
+		m_log.lock(); // safe write access for competed access from other thread
 		fprintf(options.logfd(), "%-24s", prefix);
 		vfprintf(options.logfd(), format, args);
 		fprintf(options.logfd(), "\n");
 		fflush(options.logfd());
+		m_log.unlock();
 	} else if (gStartLogBuf) {
 		char buf[500];
 		int bufUsed;
 		bufUsed = snprintf(buf, 500, "%-24s", prefix);
 		bufUsed += vsnprintf(buf + bufUsed, bufUsed < 500 ? 500 - bufUsed : 0, format, args);
 		bufUsed += snprintf(buf + bufUsed, bufUsed < 500 ? 500 - bufUsed : 0, "\n");
+		m_log.lock(); // safe write access for competed access from other thread
```

Comment:
thx. I thought std::streambuf::sputn would be thread safe but seems not (man page mentions "data races can occur".)